From 51eadb519b7a0c6e4a6c2fb7520b202f23e9e8db Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Tue, 8 Nov 2011 17:40:05 +0800
Subject: [PATCH 169/259] ENGR00161599 vpu: add O_SYNC flag for vpu pmem on
 android

Add O_SYNC flag for vpu pmem device to make cache and memory
sync.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_io.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vpu/vpu_io.c b/vpu/vpu_io.c
index 2427a57..dc52b96 100644
--- a/vpu/vpu_io.c
+++ b/vpu/vpu_io.c
@@ -337,7 +337,7 @@ int _IOGetPhyMem(int which, vpu_mem_desc *buff)
 	}
 
 	buff->size = (buff->size + pagesize-1) & ~(pagesize - 1);
-	fd_pem = (unsigned long)open("/dev/pmem_adsp", O_RDWR);
+	fd_pem = (unsigned long)open("/dev/pmem_adsp", O_RDWR | O_SYNC);
 	if (fd_pem < 0) {
 		LOGE("Error!_IOGetPhyMem Error,cannot open pmem");
 		return -1;
-- 
1.8.0

